src/NServiceBus.Core/Receiving/ReceiveComponent.Configuration.cs

```diff
@@ -33,72 +31,44 @@ public static Configuration PrepareConfiguration(Settings settings, TransportSea
         var pushRuntimeSettings = settings.PushRuntimeSettings;
 
         var receiveConfiguration = new Configuration(
-            queueNameBase,
+            new QueueAddress(queueNameBase),
             instanceSpecificQueueAddress,
             pushRuntimeSettings,
             purgeOnStartup,
             settings.PipelineCompletedSubscribers ?? new Notification<ReceivePipelineCompleted>(),
             isSendOnlyEndpoint,
             settings.ExecuteTheseHandlersFirst,
             settings.MessageHandlerRegistry,
-            settings.ShouldCreateQueues,
-            transportSeam,
-            settings.Conventions);
+            transportSeam);
 
         settings.RegisterReceiveConfigurationForBackwardsCompatibility(receiveConfiguration);
 
         return receiveConfiguration;
     }
 
-
-    public class Configuration
+    public class Configuration(QueueAddress localQueueAddress,
+        QueueAddress instanceSpecificQueueAddress,
+        PushRuntimeSettings pushRuntimeSettings,
+        bool purgeOnStartup,
+        Notification<ReceivePipelineCompleted> pipelineCompletedSubscribers,
+        bool isSendOnlyEndpoint,
+        List<Type> executeTheseHandlersFirst,
+        MessageHandlerRegistry messageHandlerRegistry,
+        TransportSeam transportSeam)
     {
-        public Configuration(string queueNameBase,
-            QueueAddress instanceSpecificQueueAddress,
-            PushRuntimeSettings pushRuntimeSettings,
-            bool purgeOnStartup,
-            Notification<ReceivePipelineCompleted> pipelineCompletedSubscribers,
-            bool isSendOnlyEndpoint,
-            List<Type> executeTheseHandlersFirst,
-            MessageHandlerRegistry messageHandlerRegistry,
-            bool createQueues,
-            TransportSeam transportSeam,
-            Conventions conventions)
-        {
-            QueueNameBase = queueNameBase;
-            LocalQueueAddress = new QueueAddress(QueueNameBase);
-            InstanceSpecificQueueAddress = instanceSpecificQueueAddress;
-            PushRuntimeSettings = pushRuntimeSettings;
-            PurgeOnStartup = purgeOnStartup;
-            IsSendOnlyEndpoint = isSendOnlyEndpoint;
-            PipelineCompletedSubscribers = pipelineCompletedSubscribers;
-            ExecuteTheseHandlersFirst = executeTheseHandlersFirst;
-            satelliteDefinitions = [];
-            this.messageHandlerRegistry = messageHandlerRegistry;
-            CreateQueues = createQueues;
-            Conventions = conventions;
-            this.transportSeam = transportSeam;
-        }
+        public QueueAddress LocalQueueAddress { get; } = localQueueAddress;
```

Comment:
There is no risk that this suddenly breaks stuff due to some assumptions in the QueueAddress class?

Reply:
All tests are green, and all 3 uses of it that I found should be fine as far as I can tell